Use public names when printing dune files

rgrinberg opened the pull request:
Fix #2425 and the bug @NathanReb encountered with @avsm.

rgrinberg committed:
Fix default implementation in dune-package

Default implementations should always use the public name

Signed-off-by: Rudi Grinberg <rudi.grinberg@gmail.com>

rgrinberg committed:
Fix implements field to always print public lib

Signed-off-by: Rudi Grinberg <rudi.grinberg@gmail.com>

rgrinberg committed:
Better formatting for library error message

Signed-off-by: Rudi Grinberg <rudi.grinberg@gmail.com>

diml commented on src/dune/lib.ml:
```diff
@@ -1755,15 +1756,36 @@ let to_dune_lib ({ name; info; _ } as lib) ~modules ~foreign_objects ~dir =
+      let open Option.O in
+      let* loc, _ = info_field in
+      let+ field = lib_field in
+      (loc, field)
```
That feels like a complicated version of `match info_field, lib_field with ...`

diml commented on test/blackbox-tests/test-cases/vlib-default-impl/run.t:
```diff
@@ -33,29 +33,15 @@ Check that ambiguity is handled correctly.
+                               ^^^^^^^^^
+  Error: Library "a-default" not found.
+  Hint: try: dune external-lib-deps --missing --root dune-package @@default
+  [1]
```
This test should be updated, otherwise it's not longer testing what it was meant to test.

rgrinberg replied:
Done.

rgrinberg committed:
Fix vlib-default-impl for dune-package

The library that was used for default-implementation didn't exist
previously and our error checking was lax.

Signed-off-by: Rudi Grinberg <rudi.grinberg@gmail.com>

rgrinberg commented:
Thanks for the review. I addressed your comments.
